### PR TITLE
Fix #1937 - add a status service

### LINKED ISSUE #1937: API Status Page
Let's do something similar to what GitHub has for its [Status Page](https://www.githubstatus.com/) with our own APIs.

Going to https://api.telescope.cdot.systems (or https://dev.api.telescope.cdot.systems or http://api.telescope.localhost) should show a static web page that lists all of our Services by name, and shows a green checkmark or red x (or something similar) indicating if the services are up or not.  This can be determined at runtime by hitting each service's `/healthcheck` endpoint.

- posts: https://dev.api.telescope.cdot.systems/v1/posts/healthcheck
- image: https://dev.api.telescope.cdot.systems/v1/image/healthcheck
- auth: https://dev.api.telescope.cdot.systems/v1/auth/healthcheck

And so on.  To make this effective, we'll also need to update a few healthcheck endpoints to include details about associated services (e.g., posts depends on redis being healthy too).

### PR DESCRIPTION
**Pull request by humphd**
This adds a basic API status service and page to help us quickly figure out if each service/web host on staging/production is working.  It looks like this right now (dev is down):

Screen Shot 2021-03-16 at 10 40 54 PM

As we add more services we can include them here.  I left a few in that aren't merged yet, just to show how it looks when things are down.

It updates every 10s or so to get current data.

### CODE REVIEW

File: src/api/status/src/server.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+service.router.use(serveStatic('public'));
+
+service.router.get('/status', (req, res) => {
+  check()
+    .then((status) => res.json(status))
+    .catch((err) => res.status(500).json({ error: err.message }));
+});
+
+const port = parseInt(process.env.IMAGE_PORT || 1111, 10);
```
Comment (manekenpix) [inline src/api/status/src/server.js:16]:
```suggestion
const port = parseInt(process.env.STATUS_PORT || 1111, 10);
```

**Review by manekenpix — CHANGES_REQUESTED** [review]

File: src/api/status/public/index.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  <body>
+    <header
+      class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-body border-bottom shadow-sm"
+    >
+      <img src="/v1/status/logo.svg" alt="Telescope Logo" width="45" height="45" />
+      <p class="h5 my-0 mx-3 me-md-auto fw-normal">Telescope API Status</p>
+      <nav class="my-2 my-md-0 me-md-3">
+        <a class="p-2 text-dark" href="https://dev.telescope.cdot.systems/">Staging</a>
+        <a class="p-2 text-dark" href="https://telescope.cdot.systems/">Production</a>
+        <a class="p-2 text-dark" href="https://github.com/Seneca-CDOT">GitHub</a>
```
Comment (manekenpix) [inline src/api/status/public/index.html:31]:
```suggestion
        <a class="p-2 text-dark" href="https://github.com/Seneca-CDOT/telescope">GitHub</a>
```

**Comment by humphd** [comment]
@manekenpix thank you for catching those mistakes.  Obviously I never use copy/paste *cough*!

I've also added a simple loading indicator on startup:

Screen Shot 2021-03-17 at 9 07 15 AM

**Comment by humphd** [comment]
Made a PR to fix the random failing e2e tests, https://github.com/Seneca-CDOT/telescope/pull/1966

**Review by HyperTHD — DISMISSED** [review]
I assume the failed e2e test was fixed with your recent merged PR, so once you rebase this, should be good to go

**Comment by humphd** [comment]
Rebased and everything is green.  Merging.

**Pull request merged**